payjoin/src/receive/v1/mod.rs

```diff
@@ -364,24 +365,28 @@ impl WantsOutputs {
     }
 }
 
-/// Shuffles `new` vector, then interleaves its elements with those from `original`,
+/// Shuffles `new` Iterator, then interleaves its elements with those from `original`,
 /// maintaining the relative order in `original` but randomly inserting elements from `new`.
 /// The combined result replaces the contents of `original`.
-fn interleave_shuffle<T: Clone, R: rand::Rng>(original: &mut Vec<T>, new: &mut [T], rng: &mut R) {
-    // Shuffle the substitute_outputs
-    new.shuffle(rng);
-    // Create a new vector to store the combined result
-    let mut combined = Vec::with_capacity(original.len() + new.len());
+fn interleave_shuffle<'a, T: Clone + 'a, R: rand::Rng, I: IntoIterator<Item = &'a T>>(
```

Comment:
Same thing applies here as well!, moreover if this is getting messy, try moving this to a `where` clause.

Reply:
We are in the era of API breaking changes before we get stable. Prefer a good design that breaks the API to keeping a crappy API stable. After 1.0 that social contract with users flips.

I agree with the `where` clause suggestion for legibility.